scripts/prepare_lambda_deployment.py

```diff
@@ -70,6 +77,56 @@ def patch_cloudwatch_resource(resource_type_name, property_name):
     return input_json
 
 
+def patch_resource_names(input_json: JSON) -> JSON:
+    # Some Chalice-generated resources have named ending in `-event`. The
+    # dash prevents generation of a fully qualified resource name (i.e.,
+    # with `config.qualified_resource_name`. Returns dict of old
+    # Terraform names and new Terraform names.
+    rename = {
+        'aws_cloudwatch_event_rule.indexercachehealth-event': 'indexercachehealth',
+        'aws_cloudwatch_event_target.indexercachehealth-event': 'indexercachehealth',
+        'aws_lambda_permission.indexercachehealth-event': 'indexercachehealth',
+        'aws_cloudwatch_event_rule.servicecachehealth-event': 'servicecachehealth',
+        'aws_cloudwatch_event_target.servicecachehealth-event': 'servicecachehealth',
+        'aws_lambda_permission.servicecachehealth-event': 'servicecachehealth',
+    }
+    replace = {k: '.'.join([k.split('.')[0], v]) for k, v in rename.items()}
+    literal = typing.TypeVar('literal', str, int)
+
+    def _patch_name(current_ref: str) -> str:
+        try:
+            new_ref = rename[current_ref]
+        except KeyError:
+            return current_ref
+        else:
+            return new_ref
+
+    def _replace_refs(value: literal) -> literal:
+        if isinstance(value, str):
+            for oldref, newref in replace.items():
+                value = value.replace(oldref, newref)
+            return value
+        else:
+            return value
```

Comment:
Is this fine written as-is or should it be

```suggestion
            for oldref, newref in replace.items():
                value = value.replace(oldref, newref)
        return value
```

The current form keeps the `else` for clarity but the proposed form is more concise.